libcache/cache.c

```diff
@@ -288,62 +287,48 @@ static cacheline_t *cache_allocateLine(cachectx_t *cache, const uint64_t setInde
 
 		for (i = 0; i < setPtr->count; ++i) {
 			if (!IS_VALID(setPtr->lines[i].flags)) {
-				lineIndex = i;
+				linePtr = &(setPtr->lines[i]);
 				break;
 			}
 		}
-
-		temp = &(setPtr->lines[lineIndex]);
-
 		for (i = 0; i < setPtr->count; ++i) {
 			if (setPtr->tags[i] == NULL) {
-				setPtr->tags[i] = temp;
+				setPtr->tags[i] = linePtr;
 				break;
 			}
 		}
 
-		LIST_ADD(&setPtr->timestamps, temp);
+		LIST_ADD(&setPtr->timestamps, linePtr);
```

Comment:
There is a possibility that we add this line to timestamps but fail on data allocation - I think we should do LIST_ADD only after we're sure that we'll succeed, so after malloc portion, along with SET_VALID